web/src/test/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/CertificateSignAndSendDescriptionFunctionImplTest.java

```diff
@@ -0,0 +1,213 @@
+import se.inera.intyg.webcert.web.service.user.dto.WebCertUser;
+import se.inera.intyg.webcert.web.web.controller.facade.dto.ResourceLinkTypeDTO;
+
+@ExtendWith(MockitoExtension.class)
+class CertificateSignAndSendDescriptionFunctionImplTest {
+
+    private final WebCertUserService webCertUserService = mock(WebCertUserService.class);
+
+    @Mock
+    private AuthoritiesHelper authoritiesHelper;
+
+    @InjectMocks
+    private CertificateSignAndSendDescriptionFunctionImpl certificateSignAndSendDescriptionFunction;
+
+    
+    @Test
+    void shallIncludeSignAndSendCertificateWithDescriptionSkatteverketWhenCertificateTypeIsDb() {
+        when(authoritiesHelper.isFeatureActive(AuthoritiesConstants.FEATURE_SIGNERA_SKICKA_DIREKT, DbModuleEntryPoint.MODULE_ID))
+            .thenReturn(true);
+        when(webCertUserService.getUser()).thenReturn(getUserWithOrigin());
+
+        final var certificate = CertificateFacadeTestHelper.createCertificate(DbModuleEntryPoint.MODULE_ID, CertificateStatus.UNSIGNED);
+        final var actualAvailableFunctions = certificateSignAndSendDescriptionFunction.get(certificate);
+
+        assertEquals(actualAvailableFunctions.get().getType(), ResourceLinkTypeDTO.SIGN_CERTIFICATE);
+        assertTrue(
+            actualAvailableFunctions.stream().anyMatch(r -> r.getDescription().contains("Intyget skickas direkt till Skatteverket.")));
+    }
+
+    @Test
+    void shallIncludeSignAndSendCertificateWithDescriptionSocialstyrelsenWhenCertificateTypeIsDoi() {
+        when(authoritiesHelper.isFeatureActive(AuthoritiesConstants.FEATURE_SIGNERA_SKICKA_DIREKT, DoiModuleEntryPoint.MODULE_ID))
+            .thenReturn(true);
+        when(webCertUserService.getUser()).thenReturn(getUserWithOrigin());
+
+        final var certificate = CertificateFacadeTestHelper.createCertificate(DoiModuleEntryPoint.MODULE_ID, CertificateStatus.UNSIGNED);
+        final var actualAvailableFunctions = certificateSignAndSendDescriptionFunction.get(certificate);
+
+        assertEquals(actualAvailableFunctions.get().getType(), ResourceLinkTypeDTO.SIGN_CERTIFICATE);
+        assertTrue(
+            actualAvailableFunctions.stream().anyMatch(r -> r.getDescription().contains("Intyget skickas direkt till Socialstyrelsen.")));
+    }
+
+    @Test
+    void shallIncludeSignAndSendCertificateWithDescriptionArbetsformedlingenWhenCertificateTypeIsAf00213() {
+        when(authoritiesHelper.isFeatureActive(AuthoritiesConstants.FEATURE_SIGNERA_SKICKA_DIREKT, Af00213EntryPoint.MODULE_ID))
+            .thenReturn(true);
+        final var certificate = CertificateFacadeTestHelper.createCertificate(Af00213EntryPoint.MODULE_ID, CertificateStatus.UNSIGNED);
+        final var actualAvailableFunctions = certificateSignAndSendDescriptionFunction.get(certificate);
+        assertEquals(actualAvailableFunctions.get().getType(), ResourceLinkTypeDTO.SIGN_CERTIFICATE);
+        assertTrue(
+            actualAvailableFunctions.stream()
```

Comment:
There's a file called ResourceLinkFacadeTestHelper where you could add this logic to check if the description matches a certain value. In that case you wouldn't have to repeat the logic and we could reuse it in the future.